rpc/src/endpoint/genesis_chunked.rs

```diff
@@ -0,0 +1,52 @@
+//! `/genesis_chunked` endpoint JSON-RPC wrapper
+
+use alloc::{string::String, vec::Vec};
+use serde::{Deserialize, Serialize};
+use tendermint_proto::serializers;
+
+use crate::{dialect::Dialect, request::RequestMessage};
+
+/// Get the genesis state in multiple chunks for the current chain
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct Request {
+    pub chunk: Option<String>,
```

Comment:
In the RPC spec, the chunk parameter is specified as an integer: https://docs.tendermint.com/v0.37/rpc/#/Info/genesis_chunked

Reply:
It was also my understanding, but when calling this query with `Option<i32>` I get:

```
outgoing request url=http://localhost:17298/ body={
  "jsonrpc": "2.0",
  "id": "93d95f7c-17ea-478a-96c4-9e11ff198ca0",
  "method": "genesis_chunked",
  "params": {
    "chunk": 0
  }
}
...
incoming response status=200 OK body={"jsonrpc":"2.0","id":"93d95f7c-17ea-478a-96c4-9e11ff198ca0","error":{"code":-32602,"message":"Invalid params","data":"error converting json params to arguments: invalid 64-bit integer encoding \"0\", expected string"}}
```

Reply:
I see, so let's take a `usize`, and convert to `Option<String>` internally

Reply:
ie. 


```rust
async fn genesis_chunked(
    &self,
    chunk: usize,
) -> Result<genesis_chunked::Response, Error> {
    self.perform(genesis_chunked::Request::new(Some(chunk.to_string()))).await
}
```